src/ingest-pipeline/airflow/dags/reorganize_upload.py

```diff
@@ -19,208 +24,373 @@
+            for lz_dir in ${WORK_DIRS[@]}; \
+            do;
+            ${PYTHON_EXE} $src_dir/metadata_extract.py --out ./${lz_dir##*/}-rslt.yml --yaml "$lz_dir" \
+              >> session.log 2> error.log ;\
+            echo $? ; \
+            if [ -s error.log ] ; \
+            then echo 'ERROR!' `cat error.log` >> session.log ; \
+            else rm error.log ; \
+            fi; \
+            done;
+            """,
+        env={
+            "AUTH_TOK": (
+                utils.get_auth_tok(
+                    **{
+                        "crypt_auth_tok": utils.encrypt_tok(
+                            airflow_conf.as_dict()["connections"]["APP_CLIENT_SECRET"]
+                        ).decode()
+                    }
+                )
+            ),
+            "PYTHON_EXE": os.environ["CONDA_PREFIX"] + "/bin/python",
+            "INGEST_API_URL": os.environ["AIRFLOW_CONN_INGEST_API_CONNECTION"],
+            "WORK_DIRS": t_split_stage_2.xcom_pull("child_work_dirs"),
+        },
+    )
 
-    t_join = JoinOperator(task_id='join')
+    t_md_consistency_tests = PythonOperator(
+        task_id="md_consistency_tests",
+        python_callable=utils.pythonop_md_consistency_tests,
+        provide_context=True,
+        op_kwargs={
+            "metadata_fname": "rslt.yml",
+            "uuid_list": t_split_stage_2.xcom_pull("child_uuid_list"),
+        },
+    )
 
+    send_status_msg = make_send_status_msg_function(
+        dag_file=__file__,
+        retcode_ops=["run_validation", "run_md_extract", "md_consistency_tests"],
+        cwl_workflows=[],
+        dataset_uuid_fun=get_dataset_uuid,
+        dataset_lz_path_fun=get_dataset_lz_path,
+        metadata_fun=read_metadata_file,
+        include_file_metadata=False,
+    )
 
-    def _get_upload_uuid(**kwargs):
-        return kwargs['ti'].xcom_pull(task_ids='find_uuid', key='uuid')
+    def wrapped_send_status_msg(**kwargs):
+        for uuid in kwargs["ti"].xcom_pull(task_ids="split_stage_2", key="child_uuid_list"):
+            if send_status_msg(uuid, get_dataset_lz_path(uuid)):
+                scanned_md = read_metadata_file(**kwargs)  # Yes, it's getting re-read
+                print(
+                    f"Got CollectionType {scanned_md['collectiontype'] if 'collectiontype' in scanned_md else None} "
+                )
+                soft_data_type = get_soft_data_type(uuid, **kwargs)
+                print(f"Got {soft_data_type} as the soft_data_type for UUID {uuid}")
+            else:
+                print(f"Something went wrong!!")
+
+    t_send_status = PythonOperator(
+        task_id="send_status_msg",
+        python_callable=wrapped_send_status_msg,
+        provide_context=True,
+        trigger_rule="all_done",
+    )
+
+    t_log_info = LogInfoOperator(task_id="log_info")
+
+    t_join = JoinOperator(task_id="join")
+
+    def flex_maybe_multiassay_spawn(**kwargs):
+        """
+        This is a generator which returns appropriate DagRunOrders
```

Comment:
This doesn't appear to be a generator any more, and it doesn't return anything, so the docstring should be fixed.

Reply:
Updated with the appropriate description